critiquebrainz/frontend/external/bookbrainz_db/literary_work.py

```diff
@@ -0,0 +1,111 @@
+import uuid
+from brainzutils import cache
+from typing import List
+import sqlalchemy
+import critiquebrainz.frontend.external.bookbrainz_db as db
+from critiquebrainz.frontend.external.bookbrainz_db import DEFAULT_CACHE_EXPIRATION
+from critiquebrainz.frontend.external.bookbrainz_db.identifiers import fetch_bb_external_identifiers
+from critiquebrainz.frontend.external.bookbrainz_db.relationships import fetch_relationships
+
+WORK_TYPE_FILTER_OPTIONS = ('Novel', 'Short Story', 'Poem')
+
+def get_literary_work_by_bbid(bbid: uuid.UUID) -> dict:
+    """
+    Get info related to a literary work using its BookBrainz ID.
+    Args:
+        bbid : BBID of the literary work.
+    Returns:
+        A dictionary containing the basic information related to the literary work.
+        It includes the following keys:
+            - bbid: BookBrainz ID of the literary work.
+            - name: Name of the literary work.
+            - sort_name: Sort name of the literary work.
+            - work_type: Type of the work.
+            - disambiguation: Disambiguation of the literary work.
+            - identifiers: A list of dictionaries containing the basic information on the identifiers.
+            - rels: A list of dictionaries containing the basic information on the relationships.
+
+        Returns None if the literary work is not found.
+    """
+    literary_work = fetch_multiple_literary_works([bbid])
+    if not literary_work:
+        return None
+    return literary_work[bbid]
+
+
+def fetch_multiple_literary_works(bbids: List[uuid.UUID], work_type=None, limit=None, offset=0) -> dict:
+    """
+    Get info related to multiple literary works using their BookBrainz IDs. 
+    Args:
+        bbids (list): List of BBID of literary works.
+        work_type (str): Filter the results by work type.
+        limit (int): Limit the number of results.
+        offset (int): Offset the results.
+    Returns:
+        A dictionary containing info of multiple literary works keyed by their BBID.
+    """
+    if bbids == []:
+        return {}
+
+    bbids = [str(uuid.UUID(bbid)) for bbid in bbids]
+
+    if limit is None:
+        limit = len(bbids)
+
+    if work_type is None:
+        work_type_filter_string = ''
+    elif work_type in WORK_TYPE_FILTER_OPTIONS:
+        work_type_filter_string = 'AND work_type = \'%s\'' % work_type
```

Comment:
even though we do some validation of this user input data, we should always use sql query parameters for anything in a WHERE statement.
We can do this by setting up the query parameters dict ahead of time:
```py
query_params = {}
elif work_type in WORK_TYPE_FILTER_OPTIONS:
     work_type_filter_string = 'AND work_type = :work_type'
     query_params['work_type'] = work_type
...
connection.execute(..., query_params)
```
You can then add additional items to query_params as needed (bbids, limit, offset, etc)

Reply:
Thats a great idea. I'll make the changes